redux_userlist_06/src/App.tsx

```diff
@@ -1,21 +1,56 @@
-import React from "react";
+import React, { useState } from "react";
+import { useEffect } from "react";
 
-import { Link, Outlet } from "react-router-dom";
+import { useDispatch, useSelector } from "react-redux";
+import axios from "axios";
+
+import { addUserList } from "../src/actions/actions";
+import { UserDetails } from "./interfaces";
+import withLoading from "./components/Dashboard/withLoading";
+import GetList from "./components/Dashboard/getList";
 
 import "./App.css";
 
+const ListWithLoading = withLoading(GetList);
+
 function App() {
+  const usersState: UserDetails[] = useSelector(
+    (state: any) => state.UserReducer
+  );
+  const dispatch = useDispatch();
+  const [isLoading,setIsLoading] = useState(true);
+
+  useEffect(() => {
+    if (usersState.length === 0) {
```

Comment:
why this check ?

Reply:
As we move back from update page to dashboard/getList , the useffect is called and data is fetched again from the api, to avoid api call i used it.